spec/tunes/app_version_spec.rb

```diff
@@ -150,6 +171,303 @@
+      end
+
+      it "can add a new trailer given a valid externally provided preview screenshot" do
+        # remove existing
+        version.upload_trailer!(nil, "English", :ipad)
+
+        count = ipad_trailers.count
+        expect(count).to eq(0)
+        version.upload_trailer!(ipad_trailer_path, "English", :ipad, '12.34', ipad_external_valid_trailer_preview_path)
+        count_after = ipad_trailers.count
+        expect(count_after ).to eq(count + 1)
+        trailer = ipad_trailers[0]
+        expect(trailer.video_asset_token).to eq('VideoSource40/v4/e3/48/1a/e3481a8f-ec25-e19f-5048-270d7acaf89a/pr_source.mov')
+        expect(trailer.picture_asset_token).to eq('Purple69/v4/5f/2b/81/5f2b814d-1083-5509-61fb-c0845f7a9374/pr_source.jpg')
+        expect(trailer.descriptionXML).to match(/FoghornLeghorn/)
+        expect(trailer.preview_frame_time_code).to eq('00:00:12:34')
+        expect(trailer.video_url).to eq(nil)
+        expect(trailer.preview_image_url).to eq(nil)
+        expect(trailer.full_sized_preview_image_url).to eq(nil)
+        expect(trailer.device_type).to eq("ipad")
+        expect(trailer.language).to eq("English")
+      end
+
+      # TODO: can we detect trailer source change ?
+
+      it "remove the video trailer" do
+        count = trailers('ipad').count
+        expect(count).to eq(1)
+        version.upload_trailer!(nil, "English", :ipad)
+        count_after = trailers('ipad').count
+        expect(count_after ).to eq(count - 1)
+      end
+    end
+
+    describe "Reading and modifying the geojson file", :du do
+      before do
+        json = JSON.parse(du_read_upload_geojson_response_success)
+        allow(client.du_client).to receive(:upload_geojson).and_return(json)
+      end
+
+      it "default geojson data is nil when value field is missing" do
+        expect(version.raw_data['transitAppFile']['value']).to eq(nil)
+        expect(version.transit_app_file).to eq(nil)
+      end
+
+      it "modifies the geojson file data after update" do
+        version.upload_geojson!(du_fixture_file_path('upload_valid.geojson'))
+        expect(version.transit_app_file.name).to eq('upload_valid.geojson')
+        expect(version.transit_app_file.url).to eq(nil)
+        expect(version.transit_app_file.asset_token).to eq('Purple1/v4/45/50/9d/45509d39-6a5d-7f55-f919-0fbc7436be61/pr_source.geojson')
+      end
+
+      it "deletes the geojson" do
+        version.upload_geojson!(du_fixture_file_path('upload_valid.geojson'))
+        version.upload_geojson!(nil)
+        expect(version.raw_data['transitAppFile']['value']).to eq(nil)
+        expect(version.transit_app_file).to eq(nil)
+      end
+    end
+
+    describe "Upload screenshots", :screenshots do
+      before do
+        allow(Spaceship::UploadFile).to receive(:from_path) do |path|
+          du_uploadimage_correct_screenshot if path == 'path_to_screenshot'
+        end
+      end
+      let(:screenshot_path) { 'path_to_screenshot' }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.